### ISSUE TITLE
Rancher Desktop fails to start on MacOS 12.3

### ISSUE THREAD

**Issue by dvu714** [rancher-sandbox/rancher-desktop#1815]
Noticed Rancher Desktop (1.1.1) now fails to start after the MacOS update to 12.3

**Steps to Reproduce**
1. Update MacOS to 12.3
2. Open/Start Rancher Desktop
3. Authorize privilege access

**Result**
Kubernetes Error
Error Starting Kubernetes
`Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1`

Last command Run:
`limactl start --tty=false 0`

Context:
`Starting virtual machine`

Some recent logfile lines:
```
time="2022-03-15T04:59:58-07:00" level=info msg="Using the existing instance \"0\""
time="2022-03-15T04:59:58-07:00" level=info msg="Starting switch daemon for \"shared\" network"
time="2022-03-15T04:59:58-07:00" level=fatal msg="\"/private/var/run/rancher-desktop-lima\" doesn't seem to be writable by the daemon (gid:1) group"
2022-03-15T11:59:58.940Z: + limactl start --tty=false 0
2022-03-15T11:59:58.940Z: Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
2022-03-15T11:59:58.940Z: Error starting lima: Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
    at ChildProcess.<anonymous> (/Applications/Rancher Desktop.app/Contents/Resources/app.asar/dist/app/background.js:1:8692)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
```

**Comment by handrei** [comment]
yes, i just updated and i'm facing the same issue

**Comment by axelspangberg** [comment]
yep I'm also having the same Issues, just updated and got the same errors as @dvu714

**Comment by alexdepalex** [comment]
The permissions on the `/private/var/run/rancher-desktop-lima` were reset to `555`.

```
dr-xr-xr-x  2 root  daemon   64 15 mrt 13:34 /private/var/run/rancher-desktop-lima
```

Workaround is resetting it to `775`.

**Comment by leandromoreira** [comment]
this seems a problem that was [solved by lima/qemu (already available)](https://github.com/lima-vm/lima/commit/a37a4bf0ac8a96dfacdde30219f4703186b0b282) but I don't know how to test it, [I found a PR to bump up lima/qemu](https://github.com/rancher-sandbox/rancher-desktop/pull/1423/files#diff-55f500d05ff6158b0c763449b648e2f54510c661fb7874aa76d9e397e3f7b109).

**Comment by rafaribe** [comment]
> The permissions on the `/private/var/run/rancher-desktop-lima` directory were reset to `555`.
> 
> ```
> dr-xr-xr-x  2 root  daemon   64 15 mrt 13:34 /private/var/run/rancher-desktop-lima
> ```
> 
> Workaround is resetting it to `775`.

@alexdepalex Thanks that works.

**Comment by bedzinsk** [comment]
To me it doesn't start even with 775 permissions:

```
➜  ~ ls -al /private/var/run/rancher-desktop-lima
total 0
dr-xr-xr-x   2 root  daemon    64 Mar 15 16:49 .
drwxrwxr-x  40 root  daemon  1280 Mar 15 16:49 ..
```

Error is same:
```
time="2022-03-15T16:49:54+02:00" level=fatal msg="\"/private/var/run/rancher-desktop-lima\" doesn't seem to be writable by the daemon (gid:1) group"
```

Tried reinstalling rancher desktop both via brew and manually but it still fails with same error

**Comment by jandubois** [comment]
We have determined the root cause of this issue:

Rancher Desktop has always erroneously created `/private/var/run/rancher-desktop-lima` with file mode `555`.

We didn't notice this because it was silently fixed by lima when it made sure the directory existed:

```console
# head -1 /etc/sudoers.d/zzzzz-rancher-desktop-lima
%everyone ALL=(root:wheel) NOPASSWD:NOSETENV: /bin/mkdir -m 775 -p /private/var/run/rancher-desktop-lima
```

Now, this is actually a bug: `mkdir -m 755 -p ...` is not supposed to change the permission bits if the directory already existed. But it did, so it covered up the bug in Rancher Desktop.

This bug has been fixed in the Monterey 12.3 release (but not in e.g. the Catalina security update).

The best workaround for now is to run this command after each reboot (`/var/run` is deleted on reboot), before you invoke Rancher Desktop the first time:

```
sudo mkdir -m 775 /private/var/run/rancher-desktop-lima
```

We are planning to make a new Rancher Desktop release next week that addresses this issue.

**Comment by th3fallen** [comment]
I'm getting that error even after chmodding that dir to 755 anyone else run into that?

**Comment by jandubois** [comment]
I've verified that current CI builds work properly on Monterey 12.3, so closing this issue now.

**Comment by jaydanielsencision** [comment]
> To me it doesn't start even with 775 permissions:
> 
> ```
> ➜  ~ ls -al /private/var/run/rancher-desktop-lima
> total 0
> dr-xr-xr-x   2 root  daemon    64 Mar 15 16:49 .
> drwxrwxr-x  40 root  daemon  1280 Mar 15 16:49 ..
> ```
> 
> Error is same:
> 
> ```
> time="2022-03-15T16:49:54+02:00" level=fatal msg="\"/private/var/run/rancher-desktop-lima\" doesn't seem to be writable by the daemon (gid:1) group"
> ```
> 
> Tried reinstalling rancher desktop both via brew and manually but it still fails with same error

Looks like the permissions applied to wrong directory, try again - should look like this instead:
```ls -al /private/var/run/rancher-desktop-lima
total 32
drwxrwxr-x   8 root    daemon     256 Mar 18 08:35 .
drwxrwxr-x  39 root    daemon    1248 Mar 18 13:26 ..

**Comment by mariusrugan** [comment]
> I've verified that current CI builds work properly on Monterey 12.3, so closing this issue now.

issue persists,
Monterey 12.3, fresh install, 

```
time="2022-03-20T22:05:12+01:00" level=info msg="Using the existing instance \"0\""
time="2022-03-20T22:05:12+01:00" level=info msg="Starting switch daemon for \"shared\" network"
time="2022-03-20T22:05:12+01:00" level=fatal msg="\"/private/var/run/rancher-desktop-lima\" doesn't seem to be writable by the daemon (gid:1) group"
2022-03-20T21:05:12.235Z: + limactl start --tty=false 0
2022-03-20T21:05:12.236Z: Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
2022-03-20T21:05:12.236Z: Error starting lima: Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
    at ChildProcess.<anonymous> (/Applications/Rancher Desktop.app/Contents/Resources/app.asar/dist/app/background.js:1:8692)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
 
```

**Comment by jandubois** [comment]
@mariusrugan Note that I said "current CI build". It cannot be fixed for the code already released, but will be fixed in the next release (later this week).

You can test the current 1.2 release candidate by downloading from https://github.com/rancher-sandbox/rancher-desktop/actions/runs/2007360261. Note that these builds are not signed, so you'll have to remove the quarantine attributes after installation.

**Comment by mariusrugan** [comment]
@jandubois thanks for reply and looking into it, also for your work, 
you are right, i misjudged the ticket being closed as "merged and build and released", sorry for that.

will wait for release.
thanks again!

**Comment by AdnanHodzic** [comment]
Running either `sudo mkdir -m 775 /private/var/run/rancher-desktop-lima` and/or `sudo chmod 775 /private/var/run/rancher-desktop-lima` does not fix the issue for me on MacOS 12.3 (Intel).

**Comment by jandubois** [comment]
@AdnanHodzic Please upgrade to RD 1.2.1; it does not have this issue and doesn't need a workaround.

**Comment by AdnanHodzic** [comment]
@jandubois I'm already running on RD 1.2.1 and as soon as it starts I run into:

Kubernetes Error
Error Starting Kubernetes
`Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1`

Last command Run:
`/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl start --tty=false 0`

Context:
`Starting virtual machine`

Some recent logfile lines:
```
time="2022-03-30T19:39:05+02:00" level=info msg="Using the existing instance \"0\""
time="2022-03-30T19:39:05+02:00" level=info msg="Starting switch daemon for \"rancher-desktop-bridged_en0\" network"
time="2022-03-30T19:39:05+02:00" level=fatal msg="failed to run [sudo --user root --group wheel --non-interactive /bin/mkdir -m 775 -p /private/var/run]: stdout=\"\", stderr=\"sudo: a password is required\\n\": exit status 1"
2022-03-30T17:39:05.764Z: + limactl start --tty=false 0
2022-03-30T17:39:05.764Z: Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
2022-03-30T17:39:05.766Z: Error starting lima: Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
    at ChildProcess.<anonymous> (/Applications/Rancher Desktop.app/Contents/Resources/app.asar/dist/app/background.js:1:8836)
    at ChildProcess.emit (node:events:394:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
```

**Comment by jandubois** [comment]
@AdnanHodzic Your issue is a duplicate of #1615 and not of this one.

This issue: `"/private/var/run/rancher-desktop-lima" doesn't seem to be writable by the daemon (gid:1) group`

Your error: `sudo: a password is required`.

**Comment by Darshan-tw** [comment]
Getting error after in **mac monterey 12.5** after launching rancher desktop. Sharing the 2 screenshots for the reference 

I tried some suggested solutions but not working 

1. Tried "sudo mkdir -m 775 /private/var/run/rancher-desktop-lima"
2. Removing the ./kube/config
![1](https://user-images.githubusercontent.com/88665864/181241779-a64b88f6-4e0c-486d-90ff-91f0f37e2001.png)
![2](https://user-images.githubusercontent.com/88665864/181241786-78ae9b62-4817-47cd-bbdb-dd49f7d0175c.png)

**Comment by remidebette** [comment]
Hi, this is still happening on 1.6.2 on MacOs Ventura 13.0.1

**Comment by jordancalder** [comment]
Same

**Comment by Darshan-tw** [comment]
The above issue got resolved ✅

- **Reason**: Mac's user name is very long **darshan.kmthoughtworks.com**

**Solution** 

- Created new user-account with shorter name issue got resolved

**Comment by Darshan-tw** [comment]
> Getting error in **mac monterey 12.5** after launching rancher desktop. Sharing the 2 screenshots for the reference
> 
> I tried some suggested solutions but not working
> 
> 1. Tried "sudo mkdir -m 775 /private/var/run/rancher-desktop-lima"
> 2. Removing the ./kube/config
> 3. Tried factory reset many times
>    ![1](https://user-images.githubusercontent.com/88665864/181241779-a64b88f6-4e0c-486d-90ff-91f0f37e2001.png)
>    ![2](https://user-images.githubusercontent.com/88665864/181241786-78ae9b62-4817-47cd-bbdb-dd49f7d0175c.png)

The above issue got resolved ✅

**Reason**: Mac's user name is very long **darshan.kmthoughtworks.com**

**Solution**
Created new user-account with shorter name issue got resolved

**Comment by namanrawal47** [comment]
I am having the exact same issue described above on Monterey 12.6.2 with Rancher Desktop 1.7.0. I don't have sudo access on my machine because it is company-owned. Are there any workarounds for this?

**Comment by Darshan-tw** [comment]
> I am having the exact same issue described above on Monterey 12.6.2 with Rancher Desktop 1.7.0. I don't have sudo access on my machine because it is company-owned. Are there any workarounds for this?
> 
> ```
> Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1
> 
> 'time="2023-02-14T17:48:05-05:00" level=fatal msg="failed to run [sudo --user root --group wheel --non-interactive /bin/mkdir -m 775 -p /private/var/run]: stdout=\\"Sudo Command Blocked: This Sudo Command has been blocked by IT Support.\\\\n\\", stderr=\\"\\": exit status 1"\n',
> ```

Afaik you need admin access. Contact ur IS team to get temporary admin access.

**Comment by jandubois** [comment]
You should really create a new Github issue instead of piling on to an old and closed one. The various comments here are about at least 4 different issues, which have been fixed, or noted workarounds. So if something doesn't work for you, you are most likely not having the same issue beyond "it doesn't work".

> Afaik you need admin access.

You should not need admin access, but you will have to disable it in the Preferences dialog. Without admin access you will not have an external IP address for the VM, but just port forwarding to the host, which is good enough most of the time.

You will also not be able to get a symlink to `/var/run/docker.sock` if you are using the moby runtime, so you will need to run `docker context use rancher-desktop` to configure the docker CLI to use the custom socket location.

**Comment by namanrawal47** [comment]
> You should really create a new Github issue instead of piling on to an old and closed one. The various comments here are about at least 4 different issues, which have been fixed, or noted workarounds. So if something doesn't work for you, you are most likely not having the same issue beyond "it doesn't work".
> 
> > Afaik you need admin access.
> 
> You should not need admin access, but you will have to disable it in the Preferences dialog. Without admin access you will not have an external IP address for the VM, but just port forwarding to the host, which is good enough most of the time.
> 
> You will also not be able to get a symlink to `/var/run/docker.sock` if you are using the moby runtime, so you will need to run `docker context use rancher-desktop` to configure the docker CLI to use the custom socket location.

I have a pending request for admin access, but that is not necessary anymore. Unchecking the admin access box and doing a factory reset solved my original problem of not being able to build a Docker image using Maven. 

I suggest more tailored logging and error messages. If Darshan-tw gets the same error
`Error: /Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl exited with code 1`
 as myself but for a different reason, it means more tailored error messages are needed.

**Comment by jandubois** [comment]
There are additional details in the log, and even in the dialog box (there is a scrollbar). The problem is many people just post the screenshot of the dialog instead of the full error message.

**Comment by erselbey** [comment]
Have you tried these?
<img width="880" alt="Screenshot 2023-11-30 at 12 50 58 PM" src="https://github.com/rancher-sandbox/rancher-desktop/assets/25445468/9fbe6c0b-7a12-4fbb-af52-a2c18de0497b">
<img width="880" alt="Screenshot 2023-11-30 at 12 51 11 PM" src="https://github.com/rancher-sandbox/rancher-desktop/assets/25445468/3d1bcfc3-f264-439f-9edd-91e941c9dab1">

**Comment by namanrawal47** [comment]
> Have you tried these? <img alt="Screenshot 2023-11-30 at 12 50 58 PM" width="880" src="https://private-user-images.githubusercontent.com/25445468/286894245-9fbe6c0b-7a12-4fbb-af52-a2c18de0497b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzNjA1MjksIm5iZiI6MTcwMTM2MDIyOSwicGF0aCI6Ii8yNTQ0NTQ2OC8yODY4OTQyNDUtOWZiZTZjMGItN2ExMi00ZmJiLWFmNTItYTJjMThkZTA0OTdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDE2MDM0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1NzA0MjdmMDJmYzZmMmZmMTE1OTgwZmE2OGNmMmFmZGNmODFjM2Q5OTRlNDE3ZTMzMjNjMDg4YWI0NmY2YzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MaEyf_MpGvVE3LAHoh_tcIITNFNr_c93qYdlaydkRcY"> <img alt="Screenshot 2023-11-30 at 12 51 11 PM" width="880" src="https://private-user-images.githubusercontent.com/25445468/286894275-3d1bcfc3-f264-439f-9edd-91e941c9dab1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzNjA1MjksIm5iZiI6MTcwMTM2MDIyOSwicGF0aCI6Ii8yNTQ0NTQ2OC8yODY4OTQyNzUtM2QxYmNmYzMtZjI2NC00MzlmLTllZGQtOTFlOTQxYzlkYWIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzExMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMTMwVDE2MDM0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyNGJiODdkNjZlODA3ZWJlMWVmNWM0ZjYzYzE0ZmUyN2RmYjU0MDg1NjIwN2ZiNjdlNzhkNzAyMWIzNDY3ZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QKrCsGjmCtUbC9r-Xs63XK2pZqJ4_MFwDZCZD0s2C_8">

I have not. I've updated macOS as well as Rancher Desktop since this issue and things have been working fine.

**Comment by RaviDasari** [comment]
@erselbey 's suggestion worked for me after macos sonoma upgrade. I have reset cache for rancher desktop and update the virtual machine settings suggested by @erselbey.
